src/pyk/cterm.py

```diff
@@ -97,6 +107,45 @@ def add_constraint(self, new_constraint: KInner) -> 'CTerm':
         return CTerm(mlAnd([self.config, new_constraint] + list(self.constraints), GENERATED_TOP_CELL))
 
 
+@dataclass(frozen=True, order=True)
+class CSubst:
+    subst: Subst
+    constraints: Tuple[KInner, ...]
+
+    def __init__(self, subst: Optional[Subst] = None, constraint: Optional[KInner] = None) -> None:
```

Comment:
Consider this more direct approach:

```suggestion
    def __init__(self, subst: Optional[Subst] = None, constraints: Iterable[KInner] = ()) -> None:
```

Then methods like `add_constraint` would be more straightforward and efficient.

Reply:
Done.

Reply:
Note that I needed to change `_is_spurious_constraint` to throw away `#Top` constraints too now.